### PR TITLE
Feature/sg 682 remove arch parms from base

### PR DESCRIPTION
**Pull request by Louis-Dupont**
Removing arch_params from base models.
The models that can be called with models.get are not modified to avoid breaking change.

### CODE REVIEW

File: src/super_gradients/training/models/classification_models/efficientnet.py [hunk 1]
```diff
@@ -411,27 +413,39 @@ class EfficientNet(SgModule):
         [1] https://arxiv.org/abs/1905.11946 (EfficientNet)
     """
 
-    def __init__(self, blocks_args=None, arch_params=None):
```
Comment (ofrimasad) [inline src/super_gradients/training/models/classification_models/efficientnet.py:414]: please fix the params in the documentation (above)

File: src/super_gradients/training/models/classification_models/efficientnet.py [hunk 2]
```diff
@@ -599,70 +609,200 @@ def get_efficientnet_params(width: float, depth: float, res: float, dropout: flo
...
+            min_depth=arch_params.min_depth,
+            depth_coefficient=arch_params.depth_coefficient,
+            dropout_rate=arch_params.dropout_rate,
+            drop_connect_rate=arch_params.drop_connect_rate,
+        )
 
 
 @register_model(Models.EFFICIENTNET_B1)
 class EfficientNetB1(EfficientNet):
     def __init__(self, arch_params):
```
Comment (ofrimasad) [inline src/super_gradients/training/models/classification_models/efficientnet.py:630]: will you add typing to all these __init__ in the next phase of the documentation fix?

File: src/super_gradients/training/models/detection_models/customizable_detector.py [hunk 3]
```diff
@@ -24,40 +24,48 @@ class CustomizableDetector(SgModule):
...
+        heads: Union[str, dict, HpmStruct, DictConfig],
+        neck: Optional[Union[str, dict, HpmStruct, DictConfig]] = None,
+        num_classes: int = None,
+        bn_eps: Optional[float] = None,
+        bn_momentum: Optional[float] = None,
+        inplace_act: Optional[bool] = True,
+        in_channels: int = 3,
+    ):
         """
         :param type_mapping: can be passed to resolve string type names in arch_params to actual types
```
Comment (ofrimasad) [inline src/super_gradients/training/models/detection_models/customizable_detector.py:39]: please fix the params in the documentation

**Review by ofrimasad — COMMENTED** [review]
In these files, many functions require adding the params to the documentation and adding typing. 
Why not fix the documentation and typing in this PR?

File: src/super_gradients/training/models/detection_models/customizable_detector.py (hunk 3 above)
Comment (Louis-Dupont) [reply src/super_gradients/training/models/detection_models/customizable_detector.py]: Added in https://github.com/Deci-AI/super-gradients/pull/760/commits/08e23d94eecbe3925473f0fcd0603ad8c63067fa

File: src/super_gradients/training/models/classification_models/efficientnet.py (hunk 2 above)
Comment (Louis-Dupont) [reply src/super_gradients/training/models/classification_models/efficientnet.py]: I added typing to arch_params, is it what you meant?93c77020d1a5cf43cd2a75f2acf506d842bf4036

**Review by ofrimasad — APPROVED** [review]
LGTM

**Pull request merged**